databus-api/src/main/java/com/bazaarvoice/emodb/databus/api/PollResult.java

```diff
@@ -19,18 +24,39 @@
  *         contain no events yet still indicate that there are more events.
  *     </li>
  * </ol>
+ *
+ * As a convenience the result has a {@link #getEvents()} method which returns all of the events from the stream in a
+ * list.  This is present primarily to support older non-streaming implementations and may be removed in a future
+ * release.
  */
 public class PollResult {
 
+    private final Iterator<Event> _eventIterator;
     private final List<Event> _events;
     private final boolean _moreEvents;
 
-    public PollResult(List<Event> events, boolean moreEvents) {
-        _events = events;
+    public PollResult(Iterator<Event> eventIterator, int approximateSize, boolean moreEvents) {
+        checkNotNull(eventIterator, "eventIterator");
+        _events = Lists.newArrayListWithCapacity(approximateSize);
         _moreEvents = moreEvents;
+
+        // Wrap the event iterator such that each event returned is appended to the event list
+        _eventIterator = Iterators.transform(eventIterator, event -> {
+            _events.add(event);
+            return event;
+        });
+    }
+
+    public Iterator<Event> getEventStream() {
+        return _eventIterator;
```

Comment:
minor: should we call this method as "getEventIterator()" as this is returning an Iterator and not a Stream, even though getEventStream() sounds appropriate. I'm saying this, as Java 8 "Stream" type is different compared to an Iterator, and user may expect to see a Stream from this method call.